### ISSUE TITLE
Classified / data defined symbology

### ISSUE THREAD

**Issue by mhugo** [Oslandia/QGeoloGIS#55]
For "regular" plots (not the special case of stratigraphy), data are rendered by considering **one** QGIS feature for the whole dataset.

It means we render a **multi**point, a **multi**linestring ... or a polygon.

The problem with this current approach is that we cannot have a symbology that changes with each data point. No classification / graduation / data defined, etc.

I am facing a use case where I need to distinguish between different kinds of data based on their attribute.

I then plan to modify the current behavior and render one QGIS feature for **each** data point. But I see the following drawbacks:
- if we want to keep the "polygon" rendering mode, we need to keep one feature = all the data points
- having one feature per data point may add some performance issues.

So, in the end, I think I will:
- add a symbology option "render each data point separately"
  - if enabled the user can choose a more complex symbology type ("classification", "graduated", "rule-based", etc.), but it only works for "points" and "lines" rendering, not "polygon"
 - if disabled (default) we keep the current rendering mode

This should however add some code complexities.

@vmora @troopa81 @delhomer Any comments ?

**Comment by delhomer** [comment]
From your summary, I understand that it is not possible to represent different data series in *the same* QGIS plot, isn't it? (Which would be equivalent to have two, maybe three, multipoints/multilines structures within the same graph.)

I'm quite uncomfortable with the idea of adding more if-then ad hoc cases, however I have no better idea, given my knowledge of the source code...

**Comment by troopa81** [comment]
@mhugo Is not the whole point of _LayerData_ [interface](https://github.com/Oslandia/QGeoloGIS/blob/master/qgeologis/data_interface.py#L66) ?

Then you can have specific symbology regarding attribute values, no?

**Comment by mhugo** [comment]
@troopa81 Yes, it is, but this is just an interface. In the end, everything is drawn as a big array of (x,y) values

**Comment by mhugo** [comment]
@delhomer Having more than one plots on the same graph would indeed answer the initial needs and should be easier (?) to implement. It also shares view with what we had in https://github.com/Oslandia/QGeoloGIS/issues/28 about "merged plots"

So a data-defined symbology for **each** plot is probably not a priority.

Thanks for your inputs !
